drivers/adc/adc_dw.c

```diff
@@ -291,24 +323,67 @@ static int adc_dw_read_request(struct device *dev, struct adc_seq_table *seq_tbl
 	return 0;
 }
 
-static int adc_dw_read(struct device *dev, struct adc_seq_table *seq_tbl)
+static int adc_dw_read(struct device *dev, const struct adc_sequence *seq_tbl)
 {
-	struct adc_info *info = dev->driver_data;
+	int ret;
 
-#ifdef CONFIG_ADC_DW_DUMMY_CONVERSION
-	if (info->dummy_conversion == ADC_NONE_DUMMY) {
-		adc_dw_read_request(dev, seq_tbl);
-		info->dummy_conversion = ADC_DONE_DUMMY;
-	}
+	adc_context_lock(&adc_info_dev.ctx, false, NULL);
+
+	ret = adc_dw_read_request(dev, seq_tbl);
+	adc_dw_disable(dev);
+	return ret;
+}
+
+#ifdef CONFIG_ADC_ASYNC
+/* Implementation of the ADC driver API function: adc_read_async. */
+static int adc_dw_read_async(struct device *dev,
+			       const struct adc_sequence *sequence,
+			       struct k_poll_signal *async)
+{
+
+	adc_context_lock(&adc_info_dev.ctx, true, async);
+	return adc_dw_read_request(dev, sequence);
+}
 #endif
-	return adc_dw_read_request(dev, seq_tbl);
+
+static void adc_context_start_sampling(struct adc_context *ctx)
+{
+	struct adc_info *info = CONTAINER_OF(ctx, struct adc_info, ctx);
+	const struct adc_config *config = info->dev->config->config_info;
+	const struct adc_sequence *entry = ctx->sequence;
+	u32_t adc_base = config->reg_base;
+	u32_t interval_us = entry->options->interval_us;
+	u32_t tmp_val;
+	u32_t ctrl;
+
+	adc_dw_enable(adc_info_dev.dev);
```

Comment:
Remove this as a consequence of removing `adc_dw_disable()`

Reply:
Ok